agent/consul/stream/event_publisher.go

```diff
@@ -23,13 +23,13 @@ type EventPublisher struct {
 	// This lock protects the topicBuffers, and snapCache
 	lock sync.RWMutex
 
-	// topicBuffers stores the head of the linked-list buffer to publish events to
+	// topicBuffers stores the head of the linked-list buffers to publish events to
 	// for a topic.
-	topicBuffers map[Topic]*eventBuffer
+	topicBuffers map[Topic]map[TopicKey]*eventBuffer
 
 	// snapCache if a cache of EventSnapshots indexed by topic and key.
 	// TODO(streaming): new snapshotCache struct for snapCache and snapCacheTTL
-	snapCache map[Topic]map[string]*eventSnapshot
+	snapCache map[Topic]map[TopicKey]*eventSnapshot
```

Comment:
Originally I was thinking that `TopicKey` would include both the Topic and the Key values, so we'd have `map[TopicKey]T`. 

I haven't thought about it too much. Do you think there are particular advantages to a two-level map over a single map? 

I thought maybe a single `map[TopicKey]T` might have been easier to manage, but maybe not.